askama_derive/src/generator.rs

```diff
@@ -638,18 +642,30 @@ impl<'a> Generator<'a> {
                             use super::input::EscapeMode::*;
                             let mut expr_buf = Buffer::new(0);
                             let wrapped = self.visit_expr(&mut expr_buf, s);
-
-                            buf_format.write("{}");
-                            buf_expr.write("&");
-                            buf_expr.write(&match (wrapped, &self.input.escaping) {
-                                (Wrapped, &Html) | (Wrapped, &None) | (Unwrapped, &None) => {
-                                    expr_buf.buf
-                                }
-                                (Unwrapped, &Html) => {
-                                    format!("::askama::MarkupDisplay::from(&{})", expr_buf.buf)
+                            if cache_id.contains_key(&expr_buf) {
```

Comment:
Let's match against `cache_id.entry()` here, which should be more efficient (single cache lookup if it is present).

Reply:
Ok